src/arch/cortex_m/tn_arch_cortex_m.S

```diff
@@ -584,7 +584,10 @@ _TN_LABEL(_tn_arch_context_switch_pend)
       ldr      r1, =ICSR_ADDR
       ldr      r0, =PENDSVSET
       str      r0, [r1]
-
+#if defined(__TN_ARCHFEAT_CORTEX_M_ARMv7M_ISA__)
```

Comment:
To make the formatting more consistent with the rest of the file, and to add more context for posterity, do you mind changing it to the following?

```c
#if defined(__TN_ARCHFEAT_CORTEX_M_ARMv7M_ISA__)
      //-- Code for Cortex-M3/M4/M4F

      //   From https://developer.arm.com/documentation/107565/0101/Memory-system/Memory-barriers/When-do-you-need-a-DSB-followed-by-an-ISB- :
      //
      //   > If a pended interrupt request needs to be recognized immediately
      //   > after being enabled in the NVIC, a DSB instruction followed by an
      //   > ISB instruction is recommended. The DSB instruction ensures that the
      //   > write to the NVIC enable register is complete, while the ISB
      //   > instruction ensures that IRQ is executed.
      //
      //   Originally we didn't have it here, but the problems were observed
      //   that sometimes the interrupt didn't fire immediately, which caused
      //   problems e.g. in `tn_eventgrp_wait` function, since it reaches the
      //   `rc = _tn_curr_run_task->task_wait_rc;` line before the interrupt
      //   fires, and we get incorrect wait result. Having these barriers fixed
      //   the issue. See https://github.com/dimonomid/tneo/pull/16 for more
      //   details.
      dsb
      isb
#endif
```

Alternatively I'll just make this change as a separate commit after merging the PR, but I thought it's cleaner if it's done as a single commit.

Reply:
I'm new to git, so I think it will be easier, if you reject the pull request and make all necessary changes. Is it ok?

Reply:
Yeah sure, I didn't want to reject the PR though just to give you some credits by merging it, but I just merged it to a temporary branch, edited it (with the authorship preserved), and merged to master as well.

Thanks again for the fix! The bug seems pretty nasty, I'll try to make a new release with it.

How did you even identify it? Sounds like it might have consumed tons of time to debug the issue, since it only reproduces "sometimes" (the worst kind of bug)

Reply:
I ported some project from cortex-m4 to cortex-m7 device. There was  some weird behaviour with "optimise for speed" option. In this project i have a task that waits for different events in different places with timeouts in loop. First wait normally exits by event and second normally exits by timeout. Sometimes the result of first wait was RC_TIMEOUT(as I found later, it was the result of second wait). I catched this situation with debugger and examined it. It took me 2 days to find and fix the problem in context switch.